scrapyd/webservice.py

```diff
@@ -82,7 +83,7 @@ def render_POST(self, txrequest):
         version = txrequest.args['version'][0]
         eggf = StringIO(txrequest.args['egg'][0])
         self.root.eggstorage.put(eggf, project, version)
-        spiders = get_spider_list(project)
+        spiders = get_spider_list(project, version=version)
```

Comment:
This was the bug that @szhem mentioned. If I add a spider while a newer project version already exists on the system, the results `get_spider_list` returns were wrong (representing the latest version).